nix/build.nix

```diff
@@ -0,0 +1,191 @@
+      for currentFile in ./*; do
+          # Extract the file name from the full path
+          filename=$(basename "$currentFile")
+          # Check if the file exists in reference folder
+          if [[ -e "${currentRepoSrc}/$filename" ]]; then
+              # Check if the content of the file in Folder A is the same as in Folder B
+              if cmp -s "$currentFile" "${currentRepoSrc}/$filename"; then
+                  rm "$currentFile"
+                  echo "Removed $filename"
+              else
+                  echo "Keeping $filename as it was modified by ./bootstrap"
+              fi
+          else
+              echo "$filename was created by ./bootstrap"
+          fi
+      done
+
+      # This should report nothing. Othewise, the Nix build will fail.
+      # grep -r /nix/store .
+
+      runHook postBuild
+    '';
+    dontPatchShebangs = true; # probalby unneeded
+    installPhase = ''
+      runHook preInstall
+      mkdir $out
+
+      cp -r . $out
+
+      runHook postInstall
+    '';
+    doCheck = false;
+    dontFixup = true;
+    # See "fixed output derivation".
+    outputHashAlgo = "sha256";
+    outputHashMode = "recursive";
+    outputHash = bootstrappedSrcHash;
+  };
+
+  # Common build dependencies apart from the compiler toolchain.
+  commonBuildDeps = [
+    autoconf
+    automake
+
+    gnumake
+    mtools
+    nasm
+    pkg-config # Checked for by ./configure but seems unused?!
+  ];
+in
+{
+  inherit bootstrappedSrc;
+  # Full build with all platforms.
+  limine = lib.makeOverridable stdenvNoCC.mkDerivation {
+    pname = "limine-dev";
+    version = "0.0.0";
+    src = bootstrappedSrc;
+    nativeBuildInputs = commonBuildDeps ++ [
+      llvmPackages.bintools
+      llvmPackages.clang
+      llvmPackages.lld
+    ];
+    configurePhase = ''
+      runHook preConfigure
+
+      # The default input source of this derivation is what we aggregated
+      # from `./bootstrap`. As this derivation holds all files but we are only
+      # interested in the ones that are not in `currentRepoSrc`, we just
+      # override all.
+      #
+      # This way we can use the actual current repo sources but still use the
+      # populated sources from the ./bootstrap script.
+      #
+      # It's very complicated, I know. But that way we can make it work, at
+      # least.
+      cp -RTf ${currentRepoSrc} .
+
+      # Extracted from ./bootstrap. To see why, check the `bootstrapedSrc`
+      # derivation.
+      #
+      # TODO, we could also do this in ./bootstrap but add a special flag.
+      autoreconf -fvi -Wall
+
+      echo Using clang:
+      clang --version
+      export TOOLCHAIN_FOR_TARGET=llvm
+      ./configure --enable-all
+
+      runHook postConfigure
+    '';
+    buildPhase = ''
+      runHook preBuild
+
+      make all -j $(nproc)
+
+      runHook postBuild
+    '';
+    installPhase = ''
+      runHook preInstall
+
+      mkdir -p $out/bin
+      cp -R bin/. $out
```

Comment:
Is there a reason that `DESTDIR=xyz make install` is not used here? Copying files manually out of `bin` as a method to perform an install isn't quite supported nor expected.

Furthermore, the command `limine --print-datadir` is hardcoded at build time to point to the datadir (aka usually `/usr/share/limine`) -- I know that nix is a bit odd when it comes to paths so I am not sure how that will play out.

Reply:
ahh, good point! The same probably applies for https://github.com/NixOS/nixpkgs/pull/298048